### PR TITLE
docs: add example of terraform-registry configuration

### PR DESCRIPTION
**Pull request by xlgmokha**
### Why:

This change introduces documentation for configuring terraform private registries for use by Dependabot.

/xref: https://github.com/github/dependabot-updates/issues/1480

### What's being changed:

A new section was added to https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates#configuration-options-for-private-registries.

Screen Shot 2021-05-26 at 12 54 03

### Check off the following:

- [x] I have reviewed my changes in staging (look for the **deploy-to-heroku** link in your pull request, then click **View deployment**).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/CONTRIBUTING.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by janiceilene** [comment]
@xlgmokha I see this is still in draft mode. Let me know when you're ready for a writer review 💖

**Comment by xlgmokha** [comment]
> I see this is still in draft mode. Let me know when you're ready for a writer review 💖

Will do. Thanks Janice.

**Comment by xlgmokha** [comment]
I think this is ready. Back to you @janiceilene.

File: content/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates.md [hunk 1]
```diff
@@ -944,3 +944,17 @@ registries:
     token: ${{secrets.MY_GITHUB_PERSONAL_TOKEN}}
 ```
 {% endraw %}
+
+#### `terraform-registry`
+
+The `terraform-registry` type supports token.
```
Comment (asciimike) [inline content/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates.md:950]:
```suggestion
The `terraform-registry` type supports a token.
```

File: content/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates.md [hunk 2]
```diff
@@ -944,3 +944,17 @@ registries:
...
+
+{% raw %}
+```yaml
+registries:
+  terraform-example:
+    type: terraform-registry
+    url: https://terraform.example.com
+    token: ${{secrets.MY_TERRAFORM_TOKEN}}
```
Comment (asciimike) [inline content/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates.md:958]:
```suggestion
    token: ${{secrets.MY_TERRAFORM_API_TOKEN}}
```

**Review by asciimike — DISMISSED** [review]
Minor changes; otherwise, LGTM

**Comment by janiceilene** [comment]
Thanks @xlgmokha! I've triaged this for writer review ⚡

**Review by mchammer01 — APPROVED** [review]
@xlgmokha - LGTM ✨

**Pull request merged**